packages/plugin-ext/src/main/node/plugin-deployer-impl.ts

```diff
@@ -114,25 +114,53 @@ export class PluginDeployerImpl implements PluginDeployer {
     }
 
     protected async deployMultipleEntries(pluginEntries: string[]): Promise<void> {
-        // resolve plugins
-        this.pluginDeployerEntries = await this.resolvePlugins(pluginEntries);
 
-        // now that we have plugins check if we have File Handler for them
-        await this.applyFileHandlers();
+        /**
+         * Iterate over all the plugins, resolving them one at a time and adding
+         * in extension dependencies
+         */
+        const futurePlugins = [];
+        while (pluginEntries.length !== 0) {
+            const currentPlugin = pluginEntries.pop() as string;
+            if (this.deployedPlugins.has(currentPlugin)) {
```

Comment:
I think it should belong to the scope of current deployment, i.e. `deployMultipleEntries`, not more.